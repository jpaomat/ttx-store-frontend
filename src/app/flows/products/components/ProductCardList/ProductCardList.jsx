// import { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ProductCard } from '../ProductCard/ProductCard';
import './ProductCardList.scss';

export const ProductCardList = ({products}) => {
	// const [products, setData] = useState(productsList);

	// useEffect(() => {
	// 	fetch('https://itx-frontend-test.onrender.com/api/product')
	// 		.then(response => response.json())
	// 		.then(json => setData(json))
	// 		.catch(error => console.error(error));
	// }, []);

	return (
		<div className='card-list'>
			{products &&
				products.map((productSelected, i) => (
					<div key={`card-list-${i}`}>
						<Link to={`/details/${productSelected.id}`}>
							<ProductCard {...productSelected} />
						</Link>
					</div>
				))}
		</div>
	);
};

ProductCardList.defaultProps = {
	products : []
};

ProductCardList.propTypes = {
	products: PropTypes.arrayOf({})
};