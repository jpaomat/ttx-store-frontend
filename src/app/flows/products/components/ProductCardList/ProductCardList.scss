@import '../../../../../assets/scss/';
.card-list {
	@include organizedColumn(space-between);
	a {
		text-decoration: auto;
	}
}

@media screen and (min-width: $break-tablet) {
	.card-list {
		@include organizedMultipleColumns(repeat(2, 1fr));
	}
}

@media screen and (min-width: $break-desktop) {
	.card-list {
		@include organizedMultipleColumns(repeat(4, 1fr));
	}
}
