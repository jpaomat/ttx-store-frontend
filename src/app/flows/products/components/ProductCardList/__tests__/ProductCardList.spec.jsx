import { ProductCardList } from '../ProductCardList';
import { MemoryRouter } from 'react-router-dom';
import { defaultMock } from './mocks';
import { render } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	// useSearchParams: () => [new URLSearchParams({ filter: 'alc' })],
	useLocation: () => jest.fn()
}));

describe('Techtronix component', () => {
	it('should render component', () => {
		const {container} = render(
			<MemoryRouter>
				<ProductCardList products={defaultMock} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
