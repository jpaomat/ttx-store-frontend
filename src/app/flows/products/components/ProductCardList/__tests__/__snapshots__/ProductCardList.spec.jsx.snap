// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Techtronix component should render component 1`] = `
<div>
  <div
    class="card-list"
  >
    <div>
      <a
        href="/details/undefinedundefined"
      >
        <div
          class="card"
        >
          <img
            class="card__image"
            src="http:image-hp-test.com"
          />
          <div
            class="card__content"
          >
            <p
              class="card__content--brand"
            >
              Acer
            </p>
            <p
              class="card__content--model"
            >
              test acer
            </p>
            <p
              class="card__content--price"
            >
              110
               €
            </p>
          </div>
          <div
            class="btn btn-container"
          >
            <button
              class="btn-ttx btn-ttx__sm"
            >
              <span>
                Agregar al carrito
              </span>
              <svg
                class="cart-icon"
                fill="currentColor"
                height="14"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 2.5A.5.5 0 0 1 .5 2H2a.5.5 0 0 1 .485.379L2.89 4H14.5a.5.5 0 0 1 .485.621l-1.5 6A.5.5 0 0 1 13 11H4a.5.5 0 0 1-.485-.379L1.61 3H.5a.5.5 0 0 1-.5-.5zM3.14 5l.5 2H5V5H3.14zM6 5v2h2V5H6zm3 0v2h2V5H9zm3 0v2h1.36l.5-2H12zm1.11 3H12v2h.61l.5-2zM11 8H9v2h2V8zM8 8H6v2h2V8zM5 8H3.89l.5 2H5V8zm0 5a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0zm9-1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/details/undefinedundefined"
      >
        <div
          class="card"
        >
          <img
            class="card__image"
            src="http:image-hp-test.com"
          />
          <div
            class="card__content"
          >
            <p
              class="card__content--brand"
            >
              Hp
            </p>
            <p
              class="card__content--model"
            >
              test hp
            </p>
            <p
              class="card__content--price"
            >
              160
               €
            </p>
          </div>
          <div
            class="btn btn-container"
          >
            <button
              class="btn-ttx btn-ttx__sm"
            >
              <span>
                Agregar al carrito
              </span>
              <svg
                class="cart-icon"
                fill="currentColor"
                height="14"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 2.5A.5.5 0 0 1 .5 2H2a.5.5 0 0 1 .485.379L2.89 4H14.5a.5.5 0 0 1 .485.621l-1.5 6A.5.5 0 0 1 13 11H4a.5.5 0 0 1-.485-.379L1.61 3H.5a.5.5 0 0 1-.5-.5zM3.14 5l.5 2H5V5H3.14zM6 5v2h2V5H6zm3 0v2h2V5H9zm3 0v2h1.36l.5-2H12zm1.11 3H12v2h.61l.5-2zM11 8H9v2h2V8zM8 8H6v2h2V8zM5 8H3.89l.5 2H5V8zm0 5a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0zm9-1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/details/undefinedundefined"
      >
        <div
          class="card"
        >
          <img
            class="card__image"
            src="http:image-hp-test.com"
          />
          <div
            class="card__content"
          >
            <p
              class="card__content--brand"
            >
              Alcatel
            </p>
            <p
              class="card__content--model"
            >
              test Alcatel
            </p>
            <p
              class="card__content--price"
            >
              120
               €
            </p>
          </div>
          <div
            class="btn btn-container"
          >
            <button
              class="btn-ttx btn-ttx__sm"
            >
              <span>
                Agregar al carrito
              </span>
              <svg
                class="cart-icon"
                fill="currentColor"
                height="14"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 2.5A.5.5 0 0 1 .5 2H2a.5.5 0 0 1 .485.379L2.89 4H14.5a.5.5 0 0 1 .485.621l-1.5 6A.5.5 0 0 1 13 11H4a.5.5 0 0 1-.485-.379L1.61 3H.5a.5.5 0 0 1-.5-.5zM3.14 5l.5 2H5V5H3.14zM6 5v2h2V5H6zm3 0v2h2V5H9zm3 0v2h1.36l.5-2H12zm1.11 3H12v2h.61l.5-2zM11 8H9v2h2V8zM8 8H6v2h2V8zM5 8H3.89l.5 2H5V8zm0 5a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0zm9-1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-2 1a2 2 0 1 1 4 0 2 2 0 0 1-4 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
